packages/core/src/agent/MessageReceiver.ts

```diff
@@ -59,21 +59,33 @@ export class MessageReceiver {
   }
 
   /**
-   * Receive and handle an inbound DIDComm message. It will unpack the message, transform it
+   * Receive and handle an inbound DIDComm message. It will decrypt the message, transform it
    * to it's corresponding message class and finally dispatch it to the dispatcher.
    *
-   * @param inboundPackedMessage the message to receive and handle
+   * @param inboundMessage the message to receive and handle
    */
-  public async receiveMessage(inboundPackedMessage: unknown, session?: TransportSession) {
-    if (typeof inboundPackedMessage !== 'object' || inboundPackedMessage == null) {
+  public async receiveMessage(inboundMessage: unknown, session?: TransportSession) {
+    if (typeof inboundMessage !== 'object' || inboundMessage == null) {
       throw new AriesFrameworkError('Invalid message received. Message should be object')
     }
 
     this.logger.debug(`Agent ${this.config.label} received message`)
 
-    const unpackedMessage = await this.unpackMessage(inboundPackedMessage as WireMessage)
-    const senderKey = unpackedMessage.senderVerkey
-    const recipientKey = unpackedMessage.recipientVerkey
+    if (this.isPlaintextMessage(inboundMessage as Record<string, unknown>)) {
```

Comment:
if you make the input to `isPlaintextMessage` unknown and do some additional checks in the method, we don't need to cast 

```suggestion
    if (this.isPlaintextMessage(inboundMessage)) {
```

Reply:
I needed to move the validation from the first line of the method inside `isPlaintextMessage`. It works now also without `as PlaintextMessage`